### PR TITLE
RadioGroup, Accordion, Datapoint, Dropdown: Fixing badge type

### PR DESCRIPTION
**Pull request by eniomoura**
#### What changed?

Adding dynamic types to hardcoded versions of BadgeType across Gestalt.

#### Why?

Adding more types to Badge would not be reflected on the BadgeType for these components.

### Links

- [Slack](https://pinterest.slack.com/archives/C13KLG5P0/p1708713858987789?thread_ts=1705437845.356239&cid=C13KLG5P0)

### CODE REVIEW

**Review by AlbertCarreras — APPROVED** [review]

File: packages/gestalt/src/Accordion.js [hunk 1]
```diff
@@ -3,16 +3,17 @@ import { type Element, type Node as ReactNode } from 'react';
...
 import Flex from './Flex';
 import IconButton from './IconButton';
 import IconButtonLink from './IconButtonLink';
 import icons from './icons/index';
 
-type BadgeType = {
+export type BadgeType = {
   text: string,
-  type?: 'info' | 'error' | 'warning' | 'success' | 'neutral' | 'darkWash' | 'lightWash',
+  type?: $ElementType<React$ElementConfig<typeof Badge>, 'type'>,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Accordion.js:16]: Screenshot by Dropbox Capture
 Let's hard type everywhere

Comment (eniomoura) [reply packages/gestalt/src/Accordion.js]: DONE

File: packages/gestalt/src/Accordion.js [hunk 2]
```diff
@@ -10,9 +10,17 @@ import IconButton from './IconButton';
 import IconButtonLink from './IconButtonLink';
 import icons from './icons/index';
 
-type BadgeType = {
+export type BadgeType = {
   text: string,
-  type?: 'info' | 'error' | 'warning' | 'success' | 'neutral' | 'darkWash' | 'lightWash',
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Accordion.js:15]: can you update the ts file as well?

Comment (eniomoura) [reply packages/gestalt/src/Accordion.js:15]: DONE

**Pull request merged**